compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -133,13 +133,16 @@ class ExtractDependencies extends Phase {
     if (depFile != null) {
       // Cannot ignore inheritance relationship coming from the same source (see sbt/zinc#417)
       def allowLocal = dep.context == DependencyByInheritance || dep.context == LocalDependencyByInheritance
-      val depClassFile =
-        if depFile.isClass then depFile
-        else depFile.resolveSibling(dep.to.binaryClassName + ".class")
-      if (depClassFile != null) {
-        // Dependency is external -- source is undefined
-        processExternalDependency(depClassFile, dep.to.binaryClassName)
-      } else if (allowLocal || depFile != sourceFile.file) {
+      if depFile.isTastyExtension then
+        val depClassFile = ctx.getSiblingClassfile(depFile)
+        if depClassFile != NoAbstractFile then
+          // did not find associated class file, e.g. for a TASTy-only classpath.
+          // The file that Zinc recieves with binaryDependency is used to lookup any either any
```

Comment:
"any either any" looks like a typo

Reply:
removed the comment since now we are always providing a classfile